basics/create-a-solver/create-a-solver.jmd

```diff
@@ -0,0 +1,205 @@
+---
+title: Creating a JSO compatible solver
+author: João Okimoto
+---
+
+![](../../jso.png)
+
+```julia
+using Pkg
+pkg"activate ."
+if false
+ pkg"add NLPModels"
+ pkg"add SolverTools"
+ pkg"add SolverBenchmark"
+ pkg"add Plots"
+ pkg"add LinearAlgebra"
+ pkg"add JSOSolvers"
+end
+pkg"instantiate"
+pkg"status"
+```
+
+This tutorial covers the basics on how to create a solver compatible with the
+JuliaSmoothOptimizers format and do basic benchmarking.
+Let's start off with a simple implementation of Newton's method:
+```julia
+using NLPModels, LinearAlgebra, SolverTools
+
+function newton(nlp :: AbstractNLPModel;
+                x :: AbstractVector = copy(nlp.meta.x0),
+                max_tol :: Real = √eps(eltype(x)),
+                max_time :: Float64 = 30.0,
+                max_iter :: Int = 100)
+
+  T = eltype(x)
+  k = 0
+  el_time = 0.0
+  start_time = time()
+  tired = el_time > max_time || k ≥ max_iter
+  optimal = norm(grad(nlp, x)) < max_tol
+
+  @info log_header([:iter, :f, :nrmgrad], [Int, T, T], hdr_override=Dict(:f => "f(x)", :nrmgrad => "‖∇f(x)‖"))
+
+  while !(optimal || tired)
+
+    fx = obj(nlp, x)
+    ∇fx = grad(nlp, x)
+    nrmgrad = norm(∇fx)
+    ∇²fx = Symmetric(hess(nlp, x), :L)
+
+    @info log_row(Any[k, fx, nrmgrad])
+
+    d = isposdef(∇²fx) ? ∇²fx\ -∇fx : -∇fx
+    t = one(T)
+    α = 0.5
+
+    while obj(nlp, x + t*d) > fx + α * ∇fx' * t * d
+      t *= 0.5
+    end
+
+    x += t*d
+
+    k += 1
+    el_time = time() - start_time
+    tired = el_time > max_time || k ≥ max_iter
+    optimal = nrmgrad < max_tol
+  end
+
+  if optimal
+    status =:first_order
+  elseif tired
+    if k ≥ max_iter
+      status =:max_iter
+    else
+      status =:max_time
+    end
+  else
+    status =:unknown
+  end
+
+  return GenericExecutionStats(status, nlp, solution=x, objective=obj(nlp, x),
+                               iter = k, elapsed_time = el_time)
+
+end
+```
+As you can see in the example above, our ```newton``` solver has a few important
+features:
+- It receives the **JSO Standard input**, an **AbtractNLPModel**, part of the [NLPModels](https://github.com/JuliaSmoothOptimizers/NLPModels.jl) package.
+- Derivatives are computed according to the **AbstractNLPModel** implementation with the ``hess`` and ``grad`` functions.
```

Comment:
implementation -> API
with the -> using the
Two ticks are for LaTeX. Use one tick.